### PR TITLE
Replace usage of nested PassManager-s in BufferizeOp::apply

### PR DESCRIPTION
**Pull request by shabalind**

### CODE REVIEW

**Comment by shabalind** [comment]
As an added bonus we get to avoid re-running cleanup transformations multiple times (might improve compilation time).

**Review by ftynse — APPROVED** [review]

File: lib/Dialect/LinalgTransform/IR/LinalgTransformOps.cpp [hunk 1]
```diff
@@ -699,17 +699,23 @@ transform::LowerVectorsOp::apply(transform::TransformResults &results,
...
-
   bufferization::OneShotBufferizationOptions options;
   options.memCpyFn = [](OpBuilder &builder, Location loc, Value from,
                         Value to) {
     return success(linalg::makeMemRefCopyOp(builder, loc, from, to));
   };
-  pm.addPass(createLinalgComprehensiveModuleBufferizePass(options));
-  if (failed(pm.run(state.getTopLevel())))
+
+  auto moduleOp = dyn_cast<ModuleOp>(state.getTopLevel());
```
Comment (ftynse) [inline lib/Dialect/LinalgTransform/IR/LinalgTransformOps.cpp:716]: Just `cast` if you don't verify the success status.

File: lib/Dialect/LinalgTransform/IR/LinalgTransformOps.cpp [hunk 2]
```diff
@@ -699,17 +699,23 @@ transform::LowerVectorsOp::apply(transform::TransformResults &results,
 // BufferizeOp
 //===---------------------------------------------------------------------===//
 
+static void applyBufferizationEnablingTransformations(ModuleOp moduleOp) {
+  RewritePatternSet patterns(moduleOp.getContext());
+  patterns.add<GeneralizePadOpPattern>(moduleOp.getContext());
+  (void)applyPatternsAndFoldGreedily(moduleOp, std::move(patterns));
```
Comment (ftynse) [inline lib/Dialect/LinalgTransform/IR/LinalgTransformOps.cpp:705]: Propagate failure maybe?

Comment (shabalind) [reply lib/Dialect/LinalgTransform/IR/LinalgTransformOps.cpp:705]: This helper comes from comprehensive bufferization pass where the failure wasn't propagated either. @matthias-springer is that intentional? Do we want to propagate failures here?

File: lib/Dialect/LinalgTransform/IR/LinalgTransformOps.cpp (hunk 1 above)
Comment (shabalind) [reply lib/Dialect/LinalgTransform/IR/LinalgTransformOps.cpp:716]: 👍

File: lib/Dialect/LinalgTransform/IR/LinalgTransformOps.cpp (hunk 2 above)
Comment (shabalind) [reply lib/Dialect/LinalgTransform/IR/LinalgTransformOps.cpp:705]: I'm keeping this as is for now, not to change the behavior compared to prior nested PassManager-based implementation. We can amend this change later if it's safe to propagate failures.

**Pull request merged**